### ISSUE TITLE
tear down of request listeners after page.close

### ISSUE THREAD

**Issue by pocketjoso** [puppeteer/puppeteer#695]
### Env
Puppeteer: 0.10.2
Node: v7.10.1
OS: macOS

### Problem
I want to avoid this error: `Protocol error (Network.continueInterceptedRequest): Target closed.`

```js
// only the relevant code
function handleInterceptedRequests (interceptedRequest) {
  // if this function is called _after_ page.close (and this happens),
  // continue below will throw, and in a way that cannot be caught here with a try/catch
  interceptedRequest.continue()
}

await page.setRequestInterceptionEnabled(true)
page.on('request', handleInterceptedRequests)

// page.close might be called BEFORE goto resolves.
// I am aware there is a timeout you can pass to the goto method;
// it does not solve my use case
setTimeout(page.close, someTimeoutValue)

await page.goto(someUrl)
```

### Suggestions
some ideas:
* Either make sure no events are sent to `request` listeners  after page.close (easiest for the consumer)
or
* make it possible to manually remove the listener
`page.removeListener('request', handleInterceptedRequests)` did not work - should it?
or
* make it possible to catch errors thrown by interceptedRequest abort/continue after page.close
```js
try {
  interceptedRequest.continue()
} catch (e) {
 // does not happen currently.
}
```

**Comment by ivibe** [comment]
page.goto(someUrl) and page.close() return Promise. To avoid errors they must be preceded with await

```
// only the relevant code
await page.setRequestInterceptionEnabled(true)
page.on('request', handleInterceptedRequests)

await page.goto(someUrl)
// BEFORE page has finished loading
await page.close()

function handleInterceptedRequests (interceptedRequest) {
  // if this function is called _after_ page.close (and this happens),
  // continue below will throw, and in a way that cannot be caught here with a try/catch
  interceptedRequest.continue()
}
```

**Comment by pocketjoso** [comment]
@ivibe I know, I just skipped it to make example clearer. Your "solution" does not address the problem this issue is about.

my actual use case is that my whole puppeteer execution is in a timeout, more like:
(still simplified - but the point is that `page.close` might happen before `page.goto` has resolved)
```js
setTimeout(page.close, someTimeoutValue)
await page.goto(someUrl)
```

I'll update the first post to show this example to avoid confusion when leaving out the `await`.

**Comment by aslushnikov** [comment]
This should be fixed with 0bea42bd8c4002ed28d100d9ccc06c6ac03a9b70

**Comment by pocketjoso** [comment]
Will test when I have a moment.~~, doesn't look like an obvious fix for this problem to me, but you know the codebase. :)~~ Nvm just saw the `starts swallow errors from Request.contine() and Request.abort().` -  Should do the trick!
